### PR TITLE
[MSUE-161]

### PR DESCRIPTION
**Pull request by Toji-Thomas**
Recreated PR #77 and #73 without conflicts.
Also added,
• array syntax changes in other files too
• added missing private properties.

### CODE REVIEW

File: lib/SiftResponse.php [hunk 1]
```diff
@@ -25,14 +25,9 @@ public function __construct($result, $httpStatusCode, $request, $curlErrno = 0,
...
-            if (function_exists('json_decode')) {
-                $this->body = json_decode($result, true);
-            } else {
-                require_once(dirname(__FILE__) . '/Services_JSON-1.0.3/JSON.php');
-                $json = new Services_JSON(SERVICES_JSON_LOOSE_TYPE);
-                $this->body = $json->decode($result);
-            }
+        if ($this->httpStatusCode >= 200 && !in_array($this->httpStatusCode, [204, 304])) {
+            
+            $this->body = json_decode($result, true);
```
Comment (vincent-sift) [inline lib/SiftResponse.php:30]: Is it possible for the json_decode method to not exist?

**Review by vincent-sift — CHANGES_REQUESTED** [review]
Can you add a test for the verifications check and resend methods, as these were referenced in the Jira ticket?

File: lib/SiftResponse.php (hunk 1 above)
Comment (Toji-Thomas) [reply lib/SiftResponse.php:30]: it is built-in in the PHP version we use.

**Comment by binish-exalture** [comment]
PR looks good and is ready of merge.

**Review by binish-exalture — APPROVED** [review]
The branch looks good and ready for merge.

**Pull request merged**